pep-0622.rst

```diff
@@ -22,88 +22,139 @@ Resolution:
 Abstract
 ========
 
-This PEP proposes to add a pattern matching statement to Python,
-inspired by similar syntax found in Scala and many other languages.
-
-The pattern syntax builds on Python’s existing syntax for sequence
-unpacking (e.g., ``a, b = value``), but is wrapped in a ``match``
-statement which compares its subject to several different “shapes”
-until one is found that fits. In addition to specifying the shape of a
-sequence to be unpacked, patterns can also specify the shape to be a
-mapping with specific keys, an instance of a given class with (optionally) specific
-attributes, a specific value, or a wildcard. Patterns can be composed
-in several ways.
-
-Syntactically, a ``match`` statement contains a *subject* expression
-and one or more ``case`` clauses, where each case clause specifies a
-pattern (the overall shape to be matched), an optional “guard” (a
-condition to be checked if the pattern matches), and a code block to
-be executed if the case clause is selected.
-
-The rest of the PEP motivates why we believe pattern matching makes a
-good addition to Python, explains our design choices, and contains a
-precise syntactic and runtime specification. We also give guidance for
-static type checkers (and one small addition to the ``typing`` module)
-and discuss the main objections and alternatives that have been
-brought up during extensive discussion of the proposal, both within
-the group of authors and in the python-dev community. Finally, we
-discuss some possible extensions that might be considered in the
-future, once the community has ample experience with the currently
-proposed syntax and semantics.
+This PEP proposes to add a **pattern matching statement** to Python,
+inspired by similar syntax found in Scala, Erlang, and other languages.
 
+Patterns and shapes
+-------------------
+
+The **pattern syntax** builds on Python’s existing syntax for sequence
+unpacking (e.g., ``a, b = value``).
+
+A ``match`` statement wraps sequence unpacking and compares the ``match``
+subject to several different **shapes** (patterns) until a **shape** fits.
```

Comment:
Here's my reworking of your suggestion, with my suggestion from above worked in.

```suggestion
A ``match`` statement compares a value (the **subject**)
to several different **shapes** (patterns) until a **shape** fits.
Each pattern describes the type and structure of the accepted values
(``a, b`` is a sequence of two elements) as well as the variables
where to capture its contents (into ``a`` and ``b``).
```

I do wonder if maybe it's better without the two parenthetical remarks:
```suggestion
A ``match`` statement compares a value (the **subject**)
to several different **shapes** (patterns) until a **shape** fits.
Each pattern describes the type and structure of the accepted values
as well as the variables where to capture its contents.
```

Reply:
:+1: